internal/core/input.rs

```diff
@@ -481,34 +496,80 @@ pub fn process_mouse_input(
     window_adapter: &Rc<dyn WindowAdapter>,
     mut mouse_input_state: MouseInputState,
 ) -> MouseInputState {
+    if matches!(mouse_event, MouseEvent::Released { .. }) {
+        mouse_input_state = process_delayed_event(window_adapter, mouse_input_state);
+    }
+
     if handle_mouse_grab(&mouse_event, window_adapter, &mut mouse_input_state) {
         return mouse_input_state;
     }
 
     let mut result = MouseInputState::default();
     let root = ItemRc::new(component.clone(), 0);
-    send_mouse_event_to_item(mouse_event, root, window_adapter, &mut result);
-
+    let r = send_mouse_event_to_item(mouse_event, root, window_adapter, &mut result, false);
+    if mouse_input_state.delayed.is_some() && !r.has_aborted() {
+        // Keep the delayed event
+        return mouse_input_state;
+    }
     send_exit_events(&mouse_input_state, mouse_event.position(), window_adapter);
 
     result
 }
 
+pub(crate) fn process_delayed_event(
+    window_adapter: &Rc<dyn WindowAdapter>,
+    mut mouse_input_state: MouseInputState,
+) -> MouseInputState {
+    let event = match mouse_input_state.delayed.take() {
+        Some(e) => e.1,
+        None => return mouse_input_state,
+    };
```

Comment:
This is where the timer is dropped and stopped, right?

Reply:
yes.
You think it sould have a comment?

Reply:
No, I think it’s fine. Just wanted to make sure I understand correctly.